### DISCUSSION TITLE
Hard to understand how globs are working

### DISCUSSION THREAD

**Discussion by kaldown** [BurntSushi/ripgrep#1584] [General] [answered]
I read [globs](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#manual-filtering-globs)

Consider the following example:
```
ps aux | grep code | tr -s ' ' | cut -d ' ' -f 11 | grep lib --glob='!*chromium*'
/usr/lib/chromium/chromium
/usr/lib/chromium/chromium
/usr/lib/chromium/chromium
/usr/lib/chromium/chromium
/usr/lib/chromium/chromium
```
```
rg --version
ripgrep 12.1.0
-SIMD -AVX (compiled)
+SIMD -AVX (runtime)

zsh --version
zsh 5.8 (x86_64-pc-linux-gnu)
```

What's the problem with the query?

**Comment by BurntSushi** [comment, marked as answer]
`--glob` only impacts recursive search. You aren't doing recursive search. You're searching stdin, assuming you have `grep` aliased to `rg`. You probably want `rg -v chromium`.

**Reply by kaldown** [reply]
Thank you. That's exactly what I'm struggling with
